### PR TITLE
API Update for HTTPMiddleware

### PR DESCRIPTION
**Pull request by tractorcow**
Merge with https://github.com/silverstripe/silverstripe-framework/pull/7052

### CODE REVIEW

File: _config/versionedrequestprocessors.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@
 Name: versionedrequestprocessors
 After:
   - 'requestprocessors'
+  - 'coresecurity'
 ---
 SilverStripe\Core\Injector\Injector:
-  VersionedRequestFilter:
-    class: SilverStripe\Versioned\VersionedRequestFilter
-  SilverStripe\Control\RequestProcessor:
+  SilverStripe\Control\Middleware\HTTPMiddleware.director:
```
Comment (sminnee) [inline _config/versionedrequestprocessors.yml]: This is now out of date syntax.

File: src/VersionedHTTPMiddleware.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+use SilverStripe\Security\Security;
+
+/**
+ * Initialises the versioned stage when a request is made.
+ */
+class VersionedHTTPMiddleware implements HTTPMiddleware
+{
+    public function process(HTTPRequest $request, callable $next)
+    {
+        // Ensure Controller::curr() is available
```
Comment (sminnee) [inline src/VersionedHTTPMiddleware.php]: Do we still need Controller::curr() to be available thanks to other refactors?

Comment (tractorcow) [inline src/VersionedHTTPMiddleware.php]: For the time being, yes, but at least it becomes less necessary.

I would remove this block at the same time as we remove controller::curr() altogether. :)

File: _config/versionedrequestprocessors.yml (hunk 1 above)
Comment (tractorcow) [inline _config/versionedrequestprocessors.yml]: Doh, I thought I updated this.

Comment (tractorcow) [inline _config/versionedrequestprocessors.yml]: Ok, pushed up fix. :)

File: src/VersionedHTTPMiddleware.php (hunk 2 above)
Comment (tractorcow) [inline src/VersionedHTTPMiddleware.php]: I think it's still necessary for security class.

**Pull request merged**